packages/babel-core/src/config/index.js

```diff
@@ -0,0 +1,30 @@
+import type Plugin from "./plugin";
+import OptionManager from "./option-manager";
+
+export type ResolvedConfig = {
+  options: Object,
+  passes: Array<Array<Plugin>>,
+};
+
+/**
+ * Standard API for loading Babel configuration data. Not for public consumption.
+ */
+export default function loadConfig(opts: Object): ResolvedConfig|null {
+  const mergedOpts = new OptionManager().init(opts);
+  if (!mergedOpts) return null;
+
+  let passes = [];
+  if (mergedOpts.plugins) {
+    passes.push(mergedOpts.plugins);
+  }
+
+  // With "passPerPreset" enabled there may still be presets in the options.
+  if (mergedOpts.presets) {
+    passes = passes.concat(mergedOpts.presets.map((preset) => preset.plugins).filter(Boolean));
```

Comment:
Yaa, neither are pretty. This will probably be removed in my rewrite anyway though.